grid/app/main/dfl/persistence/model_storage.py

```diff
@@ -0,0 +1,172 @@
+# Standard Python imports
+from typing import List, Dict, Union
+import hashlib
+
+# External imports
+from syft.serde import serialize, deserialize
+
+# Local imports
+from .database import db_instance
+from .model_cache import ModelCache
+from ...codes import MSG_FIELD
+
+
+class ModelStorage:
+    """ Manage all models hosted by an specific worker. """
+
+    def __init__(self, worker):
+        self.worker = worker
+        self.cache = ModelCache()
+
+    @property
+    def id(self) -> str:
+        """ Returns worker's id."""
+        return self.worker.id
+
+    @property
+    def models(self) -> List:
+        """ Returns a list of model ids hosted by this storage instance.
+            If persistence mode isn't enabled, it will return models stored in cache memory.
+            Returns:
+                model_ids : List of model ids hosted by this storage instance.
+        """
+        # If persistence mode enabled
+        if db_instance():
+            key = self._generate_hash_key()
+            size = db_instance().llen(key)
+            model_ids = db_instance().lrange(key, 0, size)
+            model_ids = [id.decode("utf-8") for id in model_ids]
+            return model_ids
+
+        return self.cache.models
+
+    def save_model(
+        self,
+        serialized_model: bytes,
+        model_id: str,
+        allow_download: bool,
+        allow_remote_inference: bool,
+        mpc: bool,
+    ):
```

Comment:
```suggestion
    ) -> bool:
```

I think the method should also return whether the saving is successful or not